src/bin/download_sysext.rs

```diff
@@ -114,19 +115,22 @@ impl<'a> Package<'a> {
     }
 
     fn verify_signature_on_disk(&mut self, from_path: &Path, pubkey_path: &str) -> Result<(), Box<dyn Error>> {
-        let upfile = File::open(from_path)?;
+        let mut upfile = File::open(from_path)?;
+        let mut bufdata: Vec<u8> = vec![];
 
-        // Read update payload from file, read delta update header from the payload.
-        let res_data = fs::read_to_string(from_path);
+        // Read the whole data to bufdata.
+        let _datalen = upfile.read_to_end(&mut bufdata)?;
+        upfile.rewind()?;
```

Comment:
Instead of maintaining the read position I think it's better to read the contents once and change the `read_delta_update_header` function to accept the buffer.

Reply:
All right.
I ended up creating a BufReader to pass into parsing functions, to make the code easier to maintain.